testing/src/main/java/org/motechproject/nms/testing/service/impl/TestingServiceImpl.java

```diff
@@ -244,17 +257,37 @@ public Object execute(Query query) {
                 return null;
             }
         };
-        stateDataService.executeSQLQuery(sqe);
+        try {
+            stateDataService.executeSQLQuery(sqe);
+        } catch (Exception e) {
+            String s = String.format("Exception While executing \"%s\"", sqlQuery);
+            LOGGER.error(s);
+            throw e;
+        }
     }
 
 
-    private void disableConstraints() {
-        changeConstraints(true);
-    }
+    private void truncateTable(final String table) {
+        SqlQueryExecution sqe = new SqlQueryExecution() {
 
+            @Override
+            public String getSqlQuery() {
+                return String.format("DELETE FROM %s WHERE 1=1", table);
```

Comment:
yes, but truncate was much (10x) slower with foreign keys on the table... even if the constraints were disabled